### ISSUE TITLE
Error during bulk insertion, could be about a timestamp before 1 Jan 1970

### ISSUE THREAD

**Issue by neslinesli93** [timescale/timescaledb#153]
Hi, I'm having nasty problems during bulk insertion of data inside an HyperTable (btw I'm using the [golang tool](https://github.com/timescale/timescaledb-parallel-copy) for the said insertion and it's wonderful).

When I try to insert rows which have a time before time 0 epoch (e.g. `1962-2-1T00:00:00.000Z`), it gives me this weird error:
`panic: pq: new row for relation "dimension_slice" violates check constraint "dimension_slice_range_end_check"`

This is the table structure
```
CREATE TABLE "HyperTableTest" (
          time TIMESTAMP NOT NULL,
          low DOUBLE PRECISION NULL,
          high DOUBLE PRECISION NULL,
          open DOUBLE PRECISION NULL,
          close DOUBLE PRECISION NULL,
          volume DOUBLE PRECISION NULL
        )
```

What's the deal with this? PostgreSQL supports negative timestamps, but it looks like that timescale does not...

**Comment by neslinesli93** [comment]
After the suggestions I got from slack, I've tried to put `BIGINT` as datatime instead of `TIMESTAMP`.
The result is that there is a new error on top of the old one, but I do not know whether it's related to timescale or the import tool...

The error is the following:
```
panic: pq: Should never happen: chunk not found after creation
```

**Comment by RobAtticus** [comment]
There does seem to be an error where we block ranges in the time dimensions from being less than 0. We will look into fixing this in the upcoming release.

**Comment by mfreed** [comment]
@neslinesli93 This is now supported in today's 0.4.0 release.  Please let us know if you have any other issues (or we'll close this issue out).

https://github.com/timescale/timescaledb/releases/tag/0.4.0

**Comment by neslinesli93** [comment]
I've gone vanilla postgresql, but I'll keep looking closely at this project waiting for it to get a bit more mature and tested for real case scenarios. Unfortunately, last time I tried querying a timescale db I got many errors and crashes... I will now try with real data (which has negative timestamps) and see how it goes, and open another issue if that's the case.

Thanks for the project, keep it going!
